Documentation/gettingstarted/kubeproxy-free.rst

```diff
@@ -1222,6 +1222,20 @@ working, take a look at `this KEP
     free mode, make sure that default Kubernetes services like ``kube-dns`` and ``kubernetes``
     have the required label value.
 
+Neighbor Discovery
+******************
+
+When kube-proxy replacement is enabled Cilium does L2 neighbor discovery of nodes in the cluster, in
+some setups Cilium may leave stale entries behind in the neighbor table causing ethernet frames
+between some nodes to drop. To prevent Cilium from performing this neighbor discovery and instead
+rely on the Linux kernel to discover hosts on the same L2 network you can pass the
+`--enable-l2-neigh-discovery` flag to the cilium-agent, setting it to false. Relying on the Linux
+Kernel might cause some packets to be dropped, e.g., a NodePort request can be dropped on an
+intermediate node (i.e., the one which received and is going to forward to a destination node which
+runs the selected service endpoint) if there is no L2 neigh entry in the kernel(due to the entry
+being garbage collected or that ARP resolution has not been done by the kernel), as BPF cannot drive
+the ARP resolution. 
```

Comment:
The sentence is very long, so let's break it down into two. The second:
> This is because Cilium does not drive the ARP resolution from the BPF programs.

Reply:
Addressed in https://github.com/cilium/cilium/commit/262176f973869ede633241f226d038cf60a81de4